nsdb-core/src/main/scala/io/radicalbit/nsdb/index/SimpleIndex.scala

```diff
@@ -49,14 +49,14 @@ trait SimpleIndex[T] extends Index[T] {
     * @return the query results as a list of entries.
     */
   def query[B](query: Query, fields: Seq[SimpleField], limit: Int, sort: Option[Sort])(f: T => B): Seq[B] = {
```

Comment:
since `f` name is not very explanatory we may enhance [function to obtain an element B from an element T] doc explanation. What do you think?

Reply:
It's always surprising what you might find in the depths of a considerably big codebase. 
This basically wanted to be a post-processing function. 
The idea was to give the capability to manipulate the output of the result in a parametric (and generic) fashion. Looking at the usages of this method that function is always `identity`. So rather than give it a more explanatory name and doc, I would definitely emove it.

Reply:
As per the previous comment, I wouldn't do this modification in this PR. What about in the next one that should be smaller?

Reply:
👍